lib/renderer/content-scripts-injector.js

```diff
@@ -17,12 +17,18 @@ const runContentScript = function (extensionId, url, code) {
   const context = {}
   require('@electron/internal/renderer/chrome-api').injectTo(extensionId, false, context)
   const wrapper = `((chrome) => {\n  ${code}\n  })`
-  const compiledWrapper = runInThisContext(wrapper, {
-    filename: url,
-    lineOffset: 1,
-    displayErrors: true
-  })
-  return compiledWrapper.call(this, context.chrome)
+  try {
+    const compiledWrapper = runInThisContext(wrapper, {
+      filename: url,
+      lineOffset: 1,
+      displayErrors: true
+    })
+    return compiledWrapper.call(this, context.chrome)
+  } catch (error) {
+    // TODO(samuelmaddock): Run scripts in isolated world, see chromium script_injection.cc
+    console.error(`Error running content script JavaScript for '${extensionId}'`)
+    console.error(error)
```

Comment:
An error will be thrown when a CSP is set with `'unsafe-eval'`. `runInThisContext` is a wrapper around `eval`.